skbio/sequence/_sequence.py

```diff
@@ -547,6 +547,122 @@ def observed_chars(self):
     def _string(self):
         return self._bytes.tostring()
 
+    @classmethod
+    @experimental(as_of="0.4.0-dev")
+    def concat(cls, seqs, how='strict'):
+        """ Concatenate an iterable of Sequence objects.
+
+        Parameters
+        ----------
+        seqs : iterable (Sequence)
+            An iterable of Sequence objects or appropriate subclasses.
+        how : {'strict', 'inner', 'outer'}, optional
+            How to intersect the `positional_metadata` of the sequences.
+            If 'strict': the `positional_metadata` must have the exact same
+            columns; 'inner': an inner-join of the columns; 'outer': an
+            outer-join of the columns.
+
+        Returns
+        -------
+        Sequence
+            The returned sequence will be an instance of the class which
+            called this class-method.
+
+        Raises
+        ------
+        ValueError
+            If `how` is not one of: 'strict', 'inner', or 'outer'.
+        ValueError
+            If `how` is 'strict' and the `positional_metadata` of each sequence
+            does not have the same cloumns.
+        TypeError
+            If the sequences cannot be cast as the calling class.
+
+        Examples
+        --------
+        A simple example:
```

Comment:
A little more description for each of these examples would be useful. For example:

"Concatenate DNA sequences into a new `DNA` object:"

"Concatenate DNA sequences into a `Sequence` object (type coercion):"